src/rest-server/src/middlewares/v2/protocol.js

```diff
@@ -0,0 +1,158 @@
+// Copyright (c) Microsoft Corporation
+// All rights reserved.
+//
+// MIT License
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+// documentation files (the "Software"), to deal in the Software without restriction, including without limitation
+// the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and
+// to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+// The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING
+// BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+
+// module dependencies
+const yaml = require('js-yaml');
+const mustache = require('mustache');
+const createError = require('../../util/error');
+const protocolSchema = require('../../config/v2/protocol');
+
+
+const prerequisiteTypes = [
+  'script',
+  'output',
+  'data',
+  'dockerimage',
+];
+
+const prerequisiteFields = [
+  'script',
+  'output',
+  'data',
+  'dockerImage',
+];
+
+const protocolWrap = async (req, res, next) => {
+  const protocolYAML = req.body;
+  req.rawBody = protocolYAML;
+  req.body = {
+    protocol: protocolYAML,
+  };
+  await next();
+};
+
+const protocolValidate = async (req, res, next) => {
+  const protocolYAML = req.body.protocol;
+  const protocolJSON = yaml.safeLoad(protocolYAML);
```

Comment:
In fact this is not a JSON. It's a (JavaScript) Object.

Reply:
updated